### ISSUE TITLE
(Disable) Incompatible retro extensions?

### ISSUE THREAD

**Issue by bollwyvl** [jupyterlite/jupyterlite#286]
### Problem

- `jupyterlab-tour` uses StateDB, which isn't available in (any?) of the retro pages
- also `jupyterlab-drawio` doesn't work, apparently

<details>
<summary>
These throw a bunch of scarrors:
</summary>
<pre>
Plugin '@jupyterlab/drawio-extension:plugin' failed to activate.
(anonymous) @ index.es6.js:282
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:283 Error: No provider for: @jupyterlab/filebrowser:IFileBrowserFactory.
    at f.e.resolveRequiredService (index.es6.js:197)
    at index.es6.js:155
    at Array.map (<anonymous>)
    at f.e.activatePlugin (index.es6.js:155)
    at index.es6.js:281
    at Array.map (<anonymous>)
    at f.e.start (index.es6.js:280)
    at index.js:250
(anonymous) @ index.es6.js:283
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:282 Plugin 'jupyterlab-tour:plugin' failed to activate.
(anonymous) @ index.es6.js:282
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:283 Error: No provider for: @jupyterlab/coreutils:IStateDB.
    at f.e.resolveRequiredService (index.es6.js:197)
    at index.es6.js:155
    at Array.map (<anonymous>)
    at f.e.activatePlugin (index.es6.js:155)
    at index.es6.js:281
    at Array.map (<anonymous>)
    at f.e.start (index.es6.js:280)
    at index.js:250
(anonymous) @ index.es6.js:283
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:282 Plugin 'jupyterlab-tour:user-tours' failed to activate.
(anonymous) @ index.es6.js:282
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:283 Error: No provider for: @jupyterlab/coreutils:IStateDB.
    at f.e.resolveRequiredService (index.es6.js:197)
    at index.es6.js:155
    at Array.map (<anonymous>)
    at f.e.activatePlugin (index.es6.js:155)
    at index.es6.js:281
    at Array.map (<anonymous>)
    at f.e.start (index.es6.js:280)
    at index.js:250
(anonymous) @ index.es6.js:283
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:282 Plugin 'jupyterlab-tour:default-tours' failed to activate.
(anonymous) @ index.es6.js:282
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.es6.js:283 Error: No provider for: @jupyterlab/coreutils:IStateDB.
    at f.e.resolveRequiredService (index.es6.js:197)
    at index.es6.js:155
    at Array.map (<anonymous>)
    at f.e.activatePlugin (index.es6.js:155)
    at index.es6.js:281
    at Array.map (<anonymous>)
    at f.e.start (index.es6.js:280)
    at index.js:250
(anonymous) @ index.es6.js:283
Promise.catch (async)
(anonymous) @ index.es6.js:281
e.start @ index.es6.js:280
(anonymous) @ index.js:250
async function (async)
(anonymous) @ index.js:42
60880 @ index.js:256
v @ bootstrap:19
Promise.then (async)
(anonymous) @ bootstrap.js:90
async function (async)
(anonymous) @ bootstrap.js:75
37559 @ bootstrap.js:64
v @ bootstrap:19
(anonymous) @ startup:5
(anonymous) @ bundle.js:1
index.js:251 RetroLite started, waiting for restore
index.js:253 RetroLite restored
index.es6.js:211 TypeError: Cannot read property 'model' of undefined
    at actions.js:1261
    at b (index.es6.js:2309)
    at actions.js:1260
    at Function.e [as expandParent] (actions.js:1291)
    at index.js:1016
    at l (index.es6.js:459)
    at Object.n.emit (index.es6.js:416)
    at n.emit (index.es6.js:99)
    at ee.set activeCellIndex [as activeCellIndex] (widget.js:843)
    at ee.onCellRemoved (widget.js:1341)
</pre></details>

which make it hard to find _actual_ errors.

### Proposed Solution

- add tours and drawio to `disabledExtensions` in `retro/jupyter-lite.json`

### Additional context

- observed in #283

**Comment by jtpio** [comment]
Thanks, disabling these extensions sounds good :+1: 

> `jupyterlab-tour` uses StateDB, which isn't available in (any?) of the retro pages

We could also add `StateDB` to all retro pages so `jupyterlab-tour` doesn't fail to activate.

**Comment by bollwyvl** [comment]
If it works, it'd be nice!

**Comment by jtpio** [comment]
Yeah we should be able to write tours for RetroLab that would be cool.

**Comment by bollwyvl** [comment]
so i see `statedb` in `retro/package.json`.... what's missing?

**Comment by jtpio** [comment]
Probably we can just include the `@jupyterlab/apputils-extension:state` plugin from upstream:

https://github.com/jupyterlab/jupyterlab/blob/43c18c2818eb691120d8126e6eebd169f9683c20/packages/apputils-extension/src/index.ts#L368

**Comment by jtpio** [comment]
Opened https://github.com/jupyterlite/jupyterlite/pull/287 (not tested yet, but :crossed_fingers: )

**Comment by jtpio** [comment]
> Opened #287 (not tested yet, but crossed_fingers )

Looks like that one should be enough for tour.

For drawio, we could do the changes upstream so the extension doesn't fail to activate at least (tracked in https://github.com/QuantStack/jupyterlab-drawio/issues/98)
